crates/typst/src/layout/grid/layout.rs

```diff
@@ -278,6 +373,114 @@ impl CellGrid {
     }
 }
 
+/// Given a cell's requested x and y, the vector with the resolved cell
+/// positions, the `auto_index` counter (determines the position of the next
+/// `(auto, auto)` cell) and the amount of columns in the grid, returns the
+/// final index of this cell in the vector of resolved cells.
+fn resolve_cell_position(
+    cell_x: Smart<usize>,
+    cell_y: Smart<usize>,
+    resolved_cells: &[Option<Cell>],
+    auto_index: &mut usize,
+    columns: usize,
+) -> HintedStrResult<usize> {
+    // Translates a (x, y) position to the equivalent index in the final cell vector.
+    // Errors if the position would be too large.
+    let cell_index = |x, y: usize| {
+        y.checked_mul(columns)
+            .and_then(|row_index| row_index.checked_add(x))
+            .ok_or_else(|| {
+                let position = match (cell_x, cell_y) {
+                    (Smart::Auto, Smart::Auto) => EcoString::default(),
+                    (Smart::Custom(x), Smart::Auto) => eco_format!("at column {x} "),
+                    (Smart::Auto, Smart::Custom(y)) => eco_format!("at row {y} "),
+                    (Smart::Custom(x), Smart::Custom(y)) => {
+                        eco_format!("at column {x}, row {y} ")
+                    }
+                };
+                HintedString::from(eco_format!(
+                    "the position of a cell {position}would be too large"
+                ))
```

Comment:
as far as I can tell, this really only happens for usize overflow? it's nice to handle this, but perhaps the message formatting is a bit overkill. if resolve_cell_position(..).at(..) would use the cell span instead of the grid span, then we extra x/y information isn't all that necessary.